src/plugins/auto_batch/src/sync_infer_request.cpp

```diff
@@ -5,345 +5,340 @@
-                                   const std::vector<std::shared_ptr<const ov::Node>>& outputs,
-                                   CompiledModel::WorkerInferRequest& workerRequest,
-                                   int batch_id,
-                                   int num_batch,
-                                   const std::set<std::string>& batchedInputs,
-                                   const std::set<std::string>& batchedOutputs)
-    : IInferRequestInternal(inputs, outputs),
-      m_batched_request_wrapper(workerRequest),
-      m_batch_id(batch_id),
-      m_batch_size(num_batch) {
-    ShareBlobsWithBatchRequest(batchedInputs, batchedOutputs);
-}
-
-SyncInferRequest::SyncInferRequest(const InferenceEngine::InputsDataMap& networkInputs,
-                                   const InferenceEngine::OutputsDataMap& networkOutputs,
-                                   CompiledModel::WorkerInferRequest& workerRequest,
-                                   int batch_id,
-                                   int num_batch,
-                                   const std::set<std::string>& batchedInputs,
-                                   const std::set<std::string>& batchedOutputs)
-    : IInferRequestInternal(networkInputs, networkOutputs),
+SyncInferRequest::SyncInferRequest(
+    const std::shared_ptr<const ov::autobatch_plugin::CompiledModel>& compiled_model,
+    std::shared_ptr<ov::autobatch_plugin::CompiledModel::WorkerInferRequest> workerRequest,
+    int batch_id,
+    int num_batch,
+    const std::set<std::string>& batchedInputs,
+    const std::set<std::string>& batchedOutputs)
+    : ov::ISyncInferRequest(compiled_model),
       m_batched_request_wrapper(workerRequest),
       m_batch_id(batch_id),
       m_batch_size(num_batch) {
-    ShareBlobsWithBatchRequest(batchedInputs, batchedOutputs);
+    share_tensors_with_batched_req(batchedInputs, batchedOutputs);
 }
 
-void SyncInferRequest::ShareBlobsWithBatchRequest(const std::set<std::string>& batchedInputs,
-                                                  const std::set<std::string>& batchedOutputs) {
-    // Allocate all input blobs
-    for (const auto& it : _networkInputs) {
-        auto blob = m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first);
-        InferenceEngine::Blob::Ptr res;
-        switch (it.second->getTensorDesc().getPrecision()) {
-        case InferenceEngine::Precision::FP32:
-            res = create_shared_blob_on_top_of_batched_blob<InferenceEngine::Precision::FP32>(
-                m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first),
-                it.first,
+void SyncInferRequest::share_tensors_with_batched_req(const std::set<std::string>& batchedInputs,
+                                                      const std::set<std::string>& batchedOutputs) {
+    for (const auto& it : get_inputs()) {
+        ov::Tensor res;
+        switch (it.get_tensor_ptr()->get_element_type()) {
+        case ov::element::f32:
```

Comment:
I suppose we don't need big switch / case, because new ov::Tensor does not require stored type as a template parameter